module/os/linux/spl/spl-kmem-cache.c

```diff
@@ -1018,6 +1018,14 @@ spl_cache_grow(spl_kmem_cache_t *skc, int flags, void **obj)
 	might_sleep();
 	*obj = NULL;
 
+	/*
+	 * Since we can't sleep attempt an emergency allocation to satisfy
+	 * the request.  The only alterative is to fail the allocation but
+	 * it's preferable try.  The use of KM_NOSLEEP is expected to be rare.
+	 */
+	if (flags & KM_NOSLEEP)
+		return (spl_emergency_alloc(skc, flags, obj));
```

Comment:
This should be done before the call to might_sleep(), which could be a rescheduling point.

Reply:
Indeed, fixed.